src/gwt/src/org/rstudio/studio/client/workbench/prefs/views/ConsolePreferencesPane.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * ConsolePreferencesPane.java
+ *
+ * Copyright (C) 2020 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+package org.rstudio.studio.client.workbench.prefs.views;
+
+import com.google.gwt.resources.client.ImageResource;
+import com.google.gwt.user.client.ui.Label;
+import com.google.inject.Inject;
+import org.rstudio.core.client.prefs.RestartRequirement;
+import org.rstudio.core.client.resources.ImageResource2x;
+import org.rstudio.core.client.widget.NumericValueWidget;
+import org.rstudio.core.client.widget.SelectWidget;
+import org.rstudio.studio.client.application.Desktop;
+import org.rstudio.studio.client.workbench.model.Session;
+import org.rstudio.studio.client.workbench.prefs.model.UserPrefs;
+
+
+public class ConsolePreferencesPane extends PreferencesPane
+{
+   @Inject
+   public ConsolePreferencesPane(UserPrefs prefs,
+                                 Session session,
+                                 PreferencesDialogResources res)
+   {
+      prefs_ = prefs;
+      res_ = res;
+
+      add(headerLabel("Display"));
+      add(checkboxPref("Show syntax highlighting in console input", prefs_.syntaxColorConsole()));
+      add(checkboxPref("Different color for error or message output (requires restart)", prefs_.highlightConsoleErrors()));
+      NumericValueWidget limitLengthPref =
+         numericPref("Limit length of lines to:", prefs_.consoleLineLengthLimit());
+      add(nudgeRightPlus(limitLengthPref));
+
+      consoleColorMode_ = new SelectWidget(
+         "ANSI Escape Codes:",
+         new String[] {
+            "Show ANSI colors",
+            "Remove ANSI codes",
+            "Ignore ANSI codes (1.0 behavior)"
+         },
+         new String[] {
+            UserPrefs.ANSI_CONSOLE_MODE_ON,
+            UserPrefs.ANSI_CONSOLE_MODE_STRIP,
+            UserPrefs.ANSI_CONSOLE_MODE_OFF
+         },
+         false,
+         true,
+         false);
+      add(consoleColorMode_);
+
+      // The error handler features require source references; if this R
+      // version doesn't support them, don't show these options. 
+      if (session.getSessionInfo().getHaveSrcrefAttribute())
```

Comment:
Since you're touching this code anyway, you can get rid of this check (and as much of the underlying machinery as you feel like). We no longer support any R versions that don't have source references.